src/modules/Basket/ProductsList/ProductListItem.tsx

```diff
@@ -1,33 +1,44 @@
-import { Product } from 'types'
+import { SelectedProduct } from 'types'
 import { Box, Flex, Image, Text } from '@chakra-ui/react'
 import stubImg from 'assets/img/stub.jpg'
 import CountButton from '../../../ui/CountButton'
 import closeIcon from 'assets/img/delete.svg'
 import { useBasketDispatchContext } from '../../../contexts/BasketContext'
 
 interface Props {
-  product: Product & { count: number }
+  item: SelectedProduct
 }
 
-const ProductListItem = ({ product }: Props) => {
-  const { addProduct, removeProduct, deleteProduct } =
+const ProductListItem = ({ item }: Props) => {
+  const { addProduct, removeProduct, deleteProduct, calculateDiscountedPrice } =
     useBasketDispatchContext()
+
+  const isDiscounted = Boolean(item.product.discount)
+
+  const price = isDiscounted
+    ? calculateDiscountedPrice(
+        item.product.price,
+        item.product.discount.discountPerQuantity,
+        item.count,
+      )
+    : null
+
   return (
     <Flex align="center" justify="space-between" w="100%" color="blue.200">
       <Flex gap={2} align="center">
         <Image
-          src={product.img}
+          src={item.product.img}
           boxSize={12}
           fallback={<Image boxSize={12} src={stubImg} />}
         />
 
         <Box>
           <Text maxW={130} fontSize={14} lineHeight="14px" fontWeight={600}>
-            {product.name}
+            {item.product.name}
           </Text>
           <Text fontSize={13}>
-            {Number(product.weight * product.count).toFixed(2)} gram /{' '}
-            {product.size * product.count} шт.
+            {Number(item.product.weight * item.count).toFixed(2)} gram /{' '}
```

Comment:
suggestion: calculating also better use in body of component (between states and render (return( ) )